### DISCUSSION TITLE
help with hello-f2py example, in the style of hello-cython ... so close!

### DISCUSSION THREAD

**Discussion by itcarroll** [scikit-build/scikit-build#889] [Q&A] [answered]
I am learning about Fortran extensions with skbuild, and hope my attempt will benefit other users by resulting in a PR to close scikit-build/scikit-build-sample-projects#29. I see that @frank1010111 has offered an example, but it does not use the `add_f2py_target` function that appears to be the preferred solution. Or is it? Because I'm stuck. Please take a look at itcarroll/scikit-build-sample-projects#1 if you could offer any help. Thank you!

**Comment by frank1010111** [comment, marked as answer]
I have a hypothesis that this scikit build pull request would fix our issues around getting add_f2py_target to work
https://github.com/scikit-build/scikit-build/pull/495

**Reply by itcarroll** [reply]
Thanks for the tip! Setting the `POSITION_INDEPENDENT_CODE` property on `_f2py_runtime_library` does silence one error. Still not successful end-to-end. Things to try, and then I hope I can mark this as the answer.

**Reply by itcarroll** [reply]
In fact, just adding the one line from that PR to make `_f2py_runtime_library` position independent is all I needed.

**Reply by henryiii** [reply]
Are the other changes in https://github.com/scikit-build/scikit-build/pull/495 okay? I'm not actively testing Fortran except in scikit-build-core, so am a little nervous about merging several changes. But that PR does work rebased (as I'd rather expect, as Fortran is nearly not tested or not tested!).

**Reply by itcarroll** [reply]
I can't confirm @henryiii , I did not try the whole PR ... I just borrowed the line I needed.
